src/abstractdataframe/abstractdataframe.jl

```diff
@@ -365,39 +369,58 @@ If the column does not allow missing values, `nothing` is returned.
 Consequently, `nmissing = 0` indicates that the column allows
 missing values, but does not currently contain any.
 
+Custom functions perform call `skipmissing` on columns of eltype `Union{T, Missing}`, 
+users should be aware that because of this, they cannot specify special handling
+of missing values in any of the functions given via a `Pair` argument.
+
+
 **Examples**
 
 ```julia
 df = DataFrame(i = 1:10, x = rand(10), y = rand(["a", "b", "c"], 10))
 describe(df)
 describe(df, stats = :all)
-describe(df, stats = [:min, :max])
+describe(df, :min, :max)
+describe(df, :mean, :variance => Statistics.var)
 ```
 
 """
-function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} =
-                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
-    # Check that people don't specify the wrong fields.
+StatsBase.describe(df::AbstractDataFrame) = _describe(df, [:mean, :min, :median, 
+                                                     :max, :nunique, :nmissing, 
+                                                     :eltype])
+function StatsBase.describe(df::AbstractDataFrame, stats::Union{Symbol, Pair{Symbol}}...)
+    _describe(df, collect(stats))
+end
+
+function _describe(df::AbstractDataFrame, stats::Vector)   
+    predefined_funs = Symbol[s for s in stats if s isa Symbol] 
+
     allowed_fields = [:mean, :std, :min, :q25, :median, :q75,
                       :max, :nunique, :nmissing, :first, :last, :eltype]
-    if stats == :all
-        stats = allowed_fields
-    end
 
-    if stats isa Symbol
-        if !(stats in allowed_fields)
+    if predefined_funs == [:all]
+        predefined_funs = allowed_fields
+        i = findfirst(s -> s == :all, stats)
+        splice!(stats, i, allowed_fields) # insert in the stats vector to get a good order
+    elseif :all in predefined_funs 
+            throw(ArgumentError("If the user specifies `:all` it must be the only `Symbol` argument.")) 
+    else 
+        if !issubset(predefined_funs, allowed_fields)
+            not_allowed = setdiff(predefined_funs, allowed_fields)
             allowed_msg = "\nAllowed fields are: :" * join(allowed_fields, ", :")
-            throw(ArgumentError(":$stats not allowed." * allowed_msg))
-        else
-            stats = [stats]
+            # todo: fix the printing of this
+            throw(ArgumentError("$not_allowed not allowed." * allowed_msg))
         end
     end
 
-    if !issubset(stats, allowed_fields)
-        disallowed_fields = setdiff(stats, allowed_fields)
-        allowed_msg = "\nAllowed fields are: :" * join(allowed_fields, ", :")
-        not_allowed = "Field(s) not allowed: :" * join(disallowed_fields, ", :") * "."
-        throw(ArgumentError(not_allowed * allowed_msg))
+    custom_funs = Pair[s for s in stats if s isa Pair]
+
+    ordered_names = [s isa Symbol ? s : s[1] for s in stats]
+    
+    if !allunique(ordered_names)
+        d = StatsBase.countmap(ordered_names)
```

Comment:
This is really a heavy solution. `setdiff(ordered_names, unique(ordered_names))` should be enough.

Reply:
I was thinking about this earlier - performance wise it is ~ the same 😄. But your solution is more readable.

Reply:
does this work? I don't think it does. I improved the printing though.

Reply:
OK it doesn't work. What we do currently in `names!` is `unique(nms[nonunique(DataFrame(nms=nms))])`. The reason why I'd rather avoid using `countmap` is that StatsBase is supposed to be moved to Statistics or other packages, so it may be deprecated at some point.

Reply:
used this and fixed printing.